src/kv/Scanner.cc

```diff
@@ -87,6 +87,21 @@ void Scanner::getData(Backoffer & bo)
 
         // TODO Check safe point.
 
+        // TiKV will only return locked keys if there is response level error
+        if (response->has_error())
```

Comment:
It's really amazing. Is it a new feature or has existed in TiKV for a long time? Can we add a ut for this?

Reply:
They fixed this in https://github.com/tikv/tikv/pull/9077 and https://github.com/tikv/tikv/pull/9120. You can reference this comment https://github.com/tikv/client-c/pull/56#issuecomment-732010080.

Maybe it's introduced by async commit. They were much shocked when they found this by just running TiPocket bank test, which shocked me much more since I reproduced the bug even one week earlier than them but cannot identify whether it's my fault or not. When I was running bank test with async commit enabled, there would be always 99 keys gone for a while after a few seconds without any error except lock exception.

I think we don't need to add ut for this, but need to integrate bank test with async commit enbled.

Reply:
Okay, Got it